### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by maen-al-khraisha**
- Create a class called Solver.

- Create a factorial method that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the particular case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.

- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").

- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
 When N is divisible by 3, return "fizz".
 When N is divisible by 5, return "buzz".
 When N is divisible by 3 and 5, return "fizzbuzz".
 In any other case, return N as a string (e.g. say N is 7 then return "7").

- We start with tests for all of the methods described above. The commit history should prove that we have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require './solver'
+
+describe 'Solver' do
+  context 'factorial' do
+    it 'return the factorial of the input number' do
+      number = Solver.new
+      expect(number.factorial(5)).to eq(120)
+      expect(number.factorial(0)).to eq(1)
+      expect(number.factorial(1)).to eq(1)
```
Comment (danmainah) [inline spec/solver_spec.rb:9]: - [ ] Kindly add tests for a negative value (`expect { resolver.factorial(-5) }.to raise_exception`)and special case `...
expect(resolver.factorial(1)).to eql(0)
...
` and ensure the tests pass

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @maenkhraisha and @kwekubright , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @maenkhraisha and @kwekubright,

✔️ All tests are passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
